android/src/main/java/com/reactNativeQuickActions/AppShortcutsModule.kt

```diff
@@ -0,0 +1,170 @@
+                    "AppShortcuts.popInitialAction error. " + e.message
+                )
+            )
+        }
+    }
+
+    @ReactMethod
+    @TargetApi(25)
+    fun setShortcutItems(items: ReadableArray) {
+        if (!isShortcutSupported || items.size() == 0) {
+            return
+        }
+        val currentActivity = currentActivity ?: return
+
+        val shortcutItems = fromReadableArray(items)
+        val shortcuts = shortcutItems.mapIndexed { index: Int, item: ShortcutItem ->
+            return@mapIndexed createShortcutInfo(currentActivity, id = "id$index", item)
+        }
+
+        try {
+            ShortcutManagerCompat.setDynamicShortcuts(reactApplicationContext, shortcuts)
+        } catch (error: Throwable) {
+            Log.e(LOG_TAG, "Error while setting dynamic shortcuts.", error)
+        }
+    }
+
+    private fun createShortcutInfo(activity: Activity, id: String, item: ShortcutItem): ShortcutInfoCompat {
+
+        val context = reactApplicationContext
+        val iconResId = context.resources.getIdentifier(item.icon, "drawable", context.packageName)
+        val uri: Uri? = item.userInfo?.url?.toUri()
+
+        // Class that is supposed to receive the intent when clicking the shortcut
+        val targetClass: Class<*> = getClassOrDefault(item, default = activity.javaClass)
+
+        val intent: Intent = if (uri != null) {
+            Intent(Intent.ACTION_VIEW, uri, context, targetClass)
+                .setPackage(context.packageName)
+        } else {
+            Intent(context, targetClass)
+                .setPackage(context.packageName)
+                .setAction(Intent.ACTION_VIEW)
+        }
+
+        // It is important to add the extra as PersistableBundle instead of Parcelable,
+        // because otherwise the ShortcutInfoManagerCompat throws an error when registering the shortcut.
+        intent.putExtra(SHORTCUT_ITEM, item.toPersistableBundle())
+
+        return ShortcutInfoCompat.Builder(context, id)
+            .setShortLabel(item.title)
+            .setLongLabel(item.title)
+            .setIcon(IconCompat.createWithResource(context, iconResId))
+            .setIntent(intent)
+            .build()
+    }
+
+    private fun getClassOrDefault(item: ShortcutItem, default: Class<*>): Class<*> {
+
+        val className: String = item.androidInfo?.className ?: return default
+
+        try {
+            return Class.forName(className)
+        } catch (error: ClassNotFoundException) {
+            Log.w(LOG_TAG, "Could not resolve class for name '$className'. Using default: '$default'.")
+        }
+
+        return default
+    }
+
+    @ReactMethod
+    @TargetApi(25)
+    fun clearShortcutItems() {
+        if (!isShortcutSupported) {
+            return
+        }
+        reactApplicationContext.getSystemService(ShortcutManager::class.java).removeAllDynamicShortcuts()
+    }
+
+    @ReactMethod
+    fun isSupported(callback: Callback?) {
```

Comment:
I'd prefer not to mix the callback and promise API, stick to Promise please

Reply:
@idoodler True, but I didn't change this function. It just shows up as "new" because I migrated the file from Java to Kotlin.

If we have to change this function, I will definitely migrate to Promise